support-frontend/assets/components/StripePaymentRequestButton/StripePaymentRequestButton.tsx

```diff
@@ -476,42 +473,43 @@ function PaymentRequestButton(props: PropTypes) {
 				return stripe.handleCardAction(clientSecret);
 			});
 		}
-
-		props.setPaymentRequestObject(paymentRequestObject);
 	}
 
 	useEffect(() => {
 		// Call canMakePayment on the paymentRequest object only once, once the stripe object is ready
-		if (stripe && !props.paymentRequestObject) {
+		if (stripe && props.paymentRequestObject.status === 'NOT_LOADED') {
 			initialisePaymentRequest();
 		}
 	}, [stripe, props.paymentRequestObject]);
 	useEffect(() => {
-		if (props.paymentRequestObject) {
-			void props.paymentRequestObject.canMakePayment().then((result) => {
-				if (result?.applePay) {
-					setType('APPLE_PAY');
-				} else if (result?.googlePay) {
-					setType('GOOGLE_PAY');
-				} else if (result) {
-					setType('PAY_NOW');
-				}
-			});
+		if (props.paymentRequestObject.status === 'AVAILABLE') {
+			void props.paymentRequestObject.paymentRequest
+				.canMakePayment()
+				.then((result) => {
+					if (result?.applePay) {
+						setType('APPLE_PAY');
+					} else if (result?.googlePay) {
+						setType('GOOGLE_PAY');
+					} else if (result) {
+						setType('PAY_NOW');
+					}
+				});
 		}
 	}, [props.paymentRequestObject]);
 
-	if (
-		!props.paymentRequestObject ||
-		props.stripePaymentRequestButtonData.paymentMethod === 'none'
-	) {
+	if (props.paymentRequestObject.status === 'NOT_LOADED') {
 		return null;
 	}
 
+	if (props.paymentRequestObject.status === 'NOT_AVAILABLE') {
+		return props.renderFallback ? props.renderFallback() : null;
+	}
+
```

Comment:
I tried initially just keeping the two states (`PaymentRequest | null`) but that meant even if the payment request would eventually be available, we would flash up `renderFallback` prop and then replace it with the `renderPaymentRequestButton` one. Now we can explicitly handle the three separate states.